lib/miq_environment.rb

```diff
@@ -1,6 +1,19 @@
 require 'sys-uname'
+require 'socket'
 
 module MiqEnvironment
+  # Return the fully qualified hostname for the local host.
+  #
+  def self.fully_qualified_domain_name
+    Socket.gethostbyname(Socket.gethostname).first
+  end
+
+  # Return the local IP v4 address of the local host, ignoring private addresses.
+  #
+  def self.local_ip_address
+    Socket.ip_address_list.detect { |addr| addr.ipv4? && !addr.ipv4_private? }&.ip_address
```

Comment:
@djberg96 when you have multiple loopback addresses defined this code behaves differently from the old `MiqSockUtil.getIpAddr` code

I have a pretty typical `/etc/hosts` file
```
$ cat /etc/hosts
127.0.0.1	localhost
127.0.1.1	desktop.example.com	desktop

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
```

```
>> MiqSockUtil.getIpAddr
=> "127.0.1.1"
>> MiqEnvironment.local_ip_address
=> "127.0.0.1"
```

The old code would skip `127.0.0.0/8` addresses first, then take anything where this is looking for `!ipv4_private?`
According to the docs for [Addrinfo#ipv4_private?](https://ruby-doc.org/stdlib-2.5.3/libdoc/socket/rdoc/Addrinfo.html#method-i-ipv4_private-3F) it doesn't consider `127.` addresses as private so this is always detecting those.

I think a closer analogy to what we had in MiqSockUtil would be something like:
```
ipv4addrs = Socket.ip_address_list.select(&:ipv4?)
ipv4addrs.detect { |ip| !ip.ipv4_loopback? } || ipv4addrs.first
```
Or something like that.  The old code never even distinguished between ipv4 private and not.

Reply:
And honestly anything that returns a single IP address is likely to be wrong some of the time (not a comment on this change specifically just in general with our usage of this)

Reply:
@agrare would it make sense to get all the ip addresses (maybe sliced by ipv4 and ipv6), and then use that collection to look through?